### DISCUSSION TITLE
does the api error need to be retried or may stuck in precommiting

### DISCUSSION THREAD

**Discussion by hunjixin** [filecoin-project/lotus#7801] [General Q&A] [answered]
https://github.com/filecoin-project/lotus/blob/a3c806a7a8efeb3744de69c2f3815cf9b225f7da/extern/storage-sealing/states_sealing.go#L294

**Comment by Reiers** [comment]
Hi @hunjixin 
Since its a question I will send it over to Lotus Discussion. 

Closing ticket under issues.

**Comment by LexLuthr** [comment, marked as answer]
@hunjixin The retry should be handled in the FSM. The PC1 itself is not being retried in the function but the FSM should force the retry of PC1. It can even get scheduled on a different worker. 

https://github.com/filecoin-project/lotus/blob/df7e2a884b5869e1ff37936a98157d4481263b35/extern/storage-sealing/fsm.go#L202

**Comment by LexLuthr** [comment]
I am archiving this discussion as there has been no new input from the author for more than 7 days.
